File: reactive-java/src/main/java/com/example/reactivejava/controller/EmployeeController.java
package com.example.reactivejava.controller;


import com.example.reactivejava.model.Employee;
import com.example.reactivejava.service.EmployeeService;
import reactor.core.publisher.Mono;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class EmployeeController {

	private final EmployeeService employeeService;

	EmployeeController(EmployeeService employeeService) {
		this.employeeService = employeeService;
	}

	@GetMapping(value = "/employee/{id}")
	public ResponseEntity<Mono<Employee>> employee(@PathVariable String id) {
		return ResponseEntity.ok()
				.body(employeeService.getData(id));

	}
}

File: reactive-kotlin/src/main/kotlin/com/example/reactivekotlin/controller/EmployeeController.kt
package com.example.reactivekotlin.controller

import com.example.reactivekotlin.model.Employee
import com.example.reactivekotlin.service.EmployeeService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RestController
import reactor.core.publisher.Mono

@RestController
class EmployeeController(
    private val employeeService: EmployeeService
) {

    @GetMapping(value = ["/employee/{id}"])
    fun employee(@PathVariable id: String?): ResponseEntity<Mono<Employee>> {
        return ResponseEntity.ok()
            .body(employeeService.getData(id))
    }
}